weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/SimpleEnvEntryCdiResourceInjectionProcessor.java

```diff
@@ -0,0 +1,415 @@
+ *         with an {@code @Resource} annotation, CDI is instructed to add the @Resource annotation,
+ *         thus enabling injection, as discussed in section 5.4.1.3 of the Jakarta EE 9.1 Platform specification.
+ *     </li>
+ * </ol>
+ */
+public class SimpleEnvEntryCdiResourceInjectionProcessor implements DeploymentUnitProcessor {
+
+    private static final Map<String, Class<?>> SIMPLE_ENTRY_TYPES;
+    private static final Map<Class<?>, Class<?>> PRIMITIVE_TYPES;
+
+    static {
+        Map<Class<?>, Class<?>> primitives = new HashMap<>();
+        primitives.put(byte.class, Byte.class);
+        primitives.put(short.class, Short.class);
+        primitives.put(int.class, Integer.class);
+        primitives.put(long.class, Long.class);
+        primitives.put(char.class, Character.class);
+        primitives.put(boolean.class, Boolean.class);
+        primitives.put(float.class, Float.class);
+        primitives.put(double.class, Double.class);
+        PRIMITIVE_TYPES = Collections.unmodifiableMap(primitives);
+
+        Map<String, Class<?>> types = new HashMap<>();
+        store(String.class, types);
+        store(Character.class, types);
+        store(Byte.class, types);
+        store(Short.class, types);
+        store(Integer.class, types);
+        store(Long.class, types);
+        store(Boolean.class, types);
+        store(Double.class, types);
+        store(Float.class, types);
+        store(Class.class, types);
+        primitives.forEach((k, v) -> types.put(k.getName(), v));
+        SIMPLE_ENTRY_TYPES = Collections.unmodifiableMap(types);
+    }
+
+    private static void store(Class<?> clazz, Map<String, Class<?>> map) {
+        map.put(clazz.getName(), clazz);
+    }
+
+    @Override
+    public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
+
+        final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+        final CapabilityServiceSupport support = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.CAPABILITY_SERVICE_SUPPORT);
+        final Optional<WeldCapability> optional = support.getOptionalCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class);
+        if (optional.isPresent() && optional.get().isPartOfWeldDeployment(deploymentUnit)) {
+            // See if any of the env-entry entries for this deployment require our CDI extension
+            final Module module = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+            EnvEntryCdiExtension extension = getEnvEntryCdiExtension(deploymentUnit, module.getClassLoader());
+            if (extension != null) {
+                optional.get().registerExtensionInstance(extension, deploymentUnit);
+                WeldLogger.DEPLOYMENT_LOGGER.infof("Registered CDI Extension %s", extension);
+            }
```

Comment:
Shall we log it as debug? info level may be too much and should be internationalized.

The same for other WeldLogger infof in this class.

Reply:
Thanks; I forgot to reduce those after using them to debug development.

Reply:
Fixed.